src/Control/Carrier/Fresh/Strict.hs

```diff
@@ -22,15 +23,28 @@ import Control.Monad.Trans.Class
 -- | Run a 'Fresh' effect counting up from 0.
 --
 -- @
--- 'runFresh' ('pure' a) = 'pure' a
+-- 'runFresh' n ('pure' a) = 'pure' (n, a)
 -- @
 -- @
--- 'runFresh' 'fresh' = 'pure' 0
+-- 'runFresh' n 'fresh' = 'pure' (n '+' 1, n)
 -- @
 --
 -- @since 0.1.0.0
-runFresh :: Functor m => FreshC m a -> m a
-runFresh = evalState 0 . runFreshC
+runFresh :: Int -> FreshC m a -> m (Int, a)
+runFresh n = runState n . runFreshC
```

Comment:
Do we want to provide a function that calls `runFresh 0`, to mimic the behavior of `runFresh` prior? I could go either way.

Reply:
I don’t think it’s really worth the effort.

Reply:
(Or the increase in surface area.)